sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java

```diff
@@ -0,0 +1,219 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@PrepareForTest({Handler.class})
+public class ApplicationLifecycleListenerTest {
+
+    private Handler mHandlerMock;
+    private ApplicationLifecycleListener mApplicationLifecycleListenerMock;
+    private Activity mActivityMock;
+
+    @Before
+    public void setUp() {
+        mHandlerMock = mock(Handler.class);
+        mApplicationLifecycleListenerMock = new ApplicationLifecycleListener(mHandlerMock);
+        mActivityMock = mock(Activity.class);
+        doAnswer(new Answer<Void>() {
+
+            @Override
+            public Void answer(InvocationOnMock invocation) {
+                ((Runnable) invocation.getArguments()[0]).run();
+                return null;
+            }
+        }).when(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
+        doAnswer(new Answer<Void>() {
+
+            @Override
+            public Void answer(InvocationOnMock invocation) {
+                ((Runnable) invocation.getArguments()[0]).run();
+                return null;
+            }
+        }).when(mHandlerMock).removeCallbacks(any(Runnable.class));
+    }
+
+    @After
+    public void tearDown() {
+        mHandlerMock = null;
+        mApplicationLifecycleListenerMock = null;
+        mActivityMock = null;
+    }
+
+    @Test
+    public void onActivityDoubleResumedTest() {
+
+        /* Prepare data. */
+        doAnswer(new Answer<Void>() {
+
+            @Override
+            public Void answer(final InvocationOnMock invocation) {
+
+                /* Post with a small delay to ensure that application resumes first. */
+                new Timer().schedule(new TimerTask() {
+
+                    @Override
+                    public void run() {
+                        ((Runnable) invocation.getArguments()[0]).run();
+                    }
+                }, 100);
+                return null;
+            }
+        }).when(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
+        MockCallbacks mockAppCenterService1 = mock(MockCallbacks.class);
+        MockCallbacks mockAppCenterService2 = mock(MockCallbacks.class);
+        mApplicationLifecycleListenerMock.registerApplicationLifecycleCallbacks(mockAppCenterService1);
+        mApplicationLifecycleListenerMock.registerApplicationLifecycleCallbacks(mockAppCenterService2);
+
+        /* Call onActivityResumed. */
+        mApplicationLifecycleListenerMock.onActivityResumed(mActivityMock);
+        verify(mHandlerMock, never()).removeCallbacks(any(Runnable.class));
+
+        /* Call onActivityResumed again. */
+        mApplicationLifecycleListenerMock.onActivityResumed(mActivityMock);
+        verify(mHandlerMock, never()).removeCallbacks(any(Runnable.class));
+
+        /* Pause.*/
+        mApplicationLifecycleListenerMock.onActivityPaused(mActivityMock);
+        verify(mHandlerMock, never()).postDelayed(any(Runnable.class), anyLong());
```

Comment:
Let's be more specific, we have a const that can be used instead of `anyLong`

Reply:
@annakocheshkova disagree. it's about "never" check

Reply:
Well it exists in other places, too